_posts/2019-12-03-pinot-connector.md

```diff
@@ -0,0 +1,76 @@
+---
+layout: post
+title:  "Apache Pinot Connector"
+author: Elon Azoulay
+---
+
+[Presto 328](https://prestosql.io/docs/current/release/release-328.html)
+introduces a new [Apache Pinot™ (Incubating) Connector](https://prestosql.io/docs/current/connector/pinot.html),
+which allows Presto to query data stored in [Apache Pinot™ (Incubating) Connector](https://pinot.apache.org/). 
+Not only does this allow access to pinot tables but gives users the ability to do things they could not do with pinot 
+alone such as join Pinot tables to other tables and use Presto's UDF's, window functions and complex aggregations.
+
+#Setup
+Create a properties file in the catalog directory, such as `etc/catalog/pinot.properties` which includes atleast the
+following to get started:
+
+```
+connector.name=pinot
+pinot.controller-urls=host1:9000,host2:9000
+```
+
+The `pinot.controller-urls` is a comma separated list of controller hosts. If Pinot is deployed via [Kubernetes](https://kubernetes.io/) and you expose the 
+the `pinot.controller-urls` needs to point to the controller Service endpoint. The Pinot broker and server must be accessible
+via dns as Pinot will return hostnames and not ip addresses.
+
+If you have a smaller number of Pinot servers than Presto workers or a relatively small number of rows per Pinot segment,
+you can minimize the requests to pinot by increasing the number of Pinot segments per split (default is 1 segment per split):
+
+```
+pinot.num-segments-per-split=15
+```
+
+If dns resolution is slow or you get `Request timed out` errors, you can increase the request timeout as follows:
+
+```
+pinot.request-timeout=3m
+```
+ 
+#Schema
+Pinot supports the following primitive types that do not allow null values. The corresponding Presto datatypes are:
+* Integer
+* Bigint
+* Real
+* Double
+* Varchar
+
+#Dynamic Pql
+If you would like to leverage Pinot's fast aggregation you can use a "dynamic" table where you specify the Pinot PQL 
+query as the table name:
+
+```sql
+SELECT * 
+FROM pinot.default."SELECT MAX(col1), COUNT(col2) FROM pinot_table GROUP BY col3, col4"
```

Comment:
That's cool.

Is this also in the connector's docs?

Reply:
I updated the pinot connector pull request and added to the presto-docs.